File: common/src/main/java/org/sqlunet/browser/web/WebModel.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.sqlunet.browser.web;

import com.bbou.concurrency.Task;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class WebModel extends ViewModel
{
	/**
	 * Live data
	 */
	private final MutableLiveData<String> data = new MutableLiveData<>();

	@NonNull
	public LiveData<String> getData()
	{
		return data;
	}

	public WebModel()
	{
		super();
	}

	public void loadData(@NonNull final DocumentStringLoader loader)
	{
		new Task<Void, Void, String>()
		{
			@Nullable
			@Override
			public String doJob(Void ignored)
			{
				return loader.getDoc();
			}

			@Override
			public void onDone(final String treeNode)
			{
				data.setValue(treeNode);
			}

		}.execute(null);
	}
}

File: common/src/main/java/org/sqlunet/browser/web/WebModel.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.browser.web

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.bbou.concurrency.Task

class WebModel : ViewModel() {

    private val data = MutableLiveData<String?>()

    fun getData(): LiveData<String?> {
        return data
    }

    fun loadData(loader: DocumentStringLoader) {

        object : Task<Void?, Void, String?>() {
            override fun doJob(params: Void?): String? {
                return loader.getDoc()
            }

            override fun onDone(result: String?) {
                data.value = result
            }
        }.execute(null)
    }
}
